### PR TITLE
Issue 1967

### PR DESCRIPTION
**Pull request by benrhine**
All the things that are possible with the new additions

mn create-app jpa-test -f data-hibernate-jpa
mn create-app jpa-test-mvn -f data-hibernate-jpa -b maven

mn create-app jpa-test-groovy -f data-hibernate-jpa -l groovy
mn create-app jpa-test-groovy-mvn -f data-hibernate-jpa -l groovy -b maven

mn create-app jpa-test-kotlin -f data-hibernate-jpa -l kotlin
mn create-app jpa-test-kotlin-mvn -f data-hibernate-jpa -l kotlin -b maven

// JDBC Tomcat
mn create-app jdbc-test-t -f jdbc-tomcat,data-jdbc
mn create-app jdbc-test-t-mvn -f jdbc-tomcat,data-jdbc -b maven

mn create-app jdbc-test-t-groovy -f jdbc-tomcat,data-jdbc -l groovy
mn create-app jdbc-test-t--groovy-mvn -f jdbc-tomcat,data-jdbc -l groovy -b maven

mn create-app jdbc-test-t-kotlin -f jdbc-tomcat,data-jdbc -l kotlin
mn create-app jdbc-test-t-kotlin-mvn -f jdbc-tomcat,data-jdbc -l kotlin -b maven

// JDBC Hikari
mn create-app jdbc-test-h -f jdbc-hikari,data-jdbc
mn create-app jdbc-test-h-mvn -f jdbc-hikari,data-jdbc -b maven

mn create-app jdbc-test-h-groovy -f jdbc-hikari,data-jdbc -l groovy
mn create-app jdbc-test-h-groovy-mvn -f jdbc-hikari,data-jdbc -l groovy -b maven

mn create-app jdbc-test-h-kotlin -f jdbc-hikari,data-jdbc -l kotlin
mn create-app jdbc-test-h-kotlin-mvn -f jdbc-hikari,data-jdbc -l kotlin -b maven

// create-repository test cases (Java/Groovy/Kotlin)
mn create-repository Book [jpa or JPA][jdbc or JDBC]
		
mn create-repository Book [jpa or JPA][jdbc or JDBC] (-i or --idType) Integer
		
mn create-repository Book [jpa or JPA][jdbc or JDBC] (-i or --idType) Long
		
mn create-repository Book [jpa or JPA][jdbc or JDBC] (-i or --idType) String
			
mn create-repository Book [jpa or JPA][jdbc or JDBC] (-a or --annotationType)

### CODE REVIEW

File: base/features/data-hibernate-jpa/skeleton/src/main/resources/application.yml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+datasources:
+  default:
+    url: jdbc:h2:mem:devDb;MVCC=TRUE;LOCK_TIMEOUT=10000;DB_CLOSE_ON_EXIT=FALSE
+    driverClassName: org.h2.Driver
+    username: sa
+    password: ''
+    schema-generate: CREATE_DROP
```
Comment (graemerocher) [inline base/features/data-hibernate-jpa/skeleton/src/main/resources/application.yml]: This applies only to the JDBC impl

File: base/features/kotlin/skeleton/gradle-build/gradle.properties [hunk 2]
```diff
@@ -1,2 +1,3 @@
 micronautVersion=@version@
+micronautDataVersion=1.0.0.M4
```
Comment (graemerocher) [inline base/features/kotlin/skeleton/gradle-build/gradle.properties]: latest version is 1.0.0.M5 and I don't think we need this here

File: base/features/kotlin/skeleton/maven-build/pom.xml [hunk 3]
```diff
@@ -1,6 +1,7 @@
 <project>
   <properties>
     <kotlinVersion>1.3.50</kotlinVersion>
+    <micronautDataVersion>1.0.0.M4</micronautDataVersion>
```
Comment (graemerocher) [inline base/features/kotlin/skeleton/maven-build/pom.xml]: Please remove this

**Comment by graemerocher** [comment]
Before merging this we need to create a 1.2.x branch

File: base/features/data-hibernate-jpa/skeleton/src/main/resources/application.yml (hunk 1 above)
Comment (benrhine) [reply base/features/data-hibernate-jpa/skeleton/src/main/resources/application.yml]: removed

File: base/features/kotlin/skeleton/gradle-build/gradle.properties (hunk 2 above)
Comment (benrhine) [reply base/features/kotlin/skeleton/gradle-build/gradle.properties]: this was a leftover from something i tried earlier i missed, it has been removed

File: base/features/kotlin/skeleton/maven-build/pom.xml (hunk 3 above)
Comment (benrhine) [reply base/features/kotlin/skeleton/maven-build/pom.xml]: this was a leftover from something i tried earlier i missed, it has been removed

File: base/features/data-hibernate-jpa/skeleton/src/main/resources/application.yml (hunk 1 above)
Comment (benrhine) [reply base/features/data-hibernate-jpa/skeleton/src/main/resources/application.yml]: this was a leftover from something i tried earlier i missed, it has been removed

**Comment by benrhine** [comment]
> Before merging this we need to create a 1.2.x branch

What exactly does this entail? @jameskleeh what exactly needs to happen for this? Thanks

**Comment by benrhine** [comment]
I also believe this pr will take of this other issue / request as well https://github.com/micronaut-projects/micronaut-profiles/issues/184

File: base/features/data-hibernate-jpa/skeleton/src/main/resources/application.yml [hunk 4]
```diff
@@ -1,11 +1,3 @@
-datasources:
-  default:
-    url: jdbc:h2:mem:devDb;MVCC=TRUE;LOCK_TIMEOUT=10000;DB_CLOSE_ON_EXIT=FALSE
-    driverClassName: org.h2.Driver
-    username: sa
-    password: ''
-    schema-generate: CREATE_DROP
```
Comment (graemerocher) [inline base/features/data-hibernate-jpa/skeleton/src/main/resources/application.yml]: I think you misunderstood sorry. Only `schema-generate` should have been deleted

Comment (benrhine) [reply base/features/data-hibernate-jpa/skeleton/src/main/resources/application.yml]: I did misunderstand, I have put everything back other than the schema-generate

**Pull request merged**